src/Http/Http.Extensions/gen/StaticRouteHandlerModel/StaticRouteHandlerModel.Emitter.cs

```diff
@@ -159,8 +159,12 @@ private static string EmitResponseWritingCall(this Endpoint endpoint)
     public static string EmitFilteredRequestHandler(this Endpoint endpoint, int baseIndent = 0)
     {
         var argumentList = endpoint.Parameters.Length == 0 ? string.Empty : $", {endpoint.EmitArgumentList()}";
-        var invocationConstructor = endpoint.Parameters.Length == 0 || endpoint.Parameters.Length > 8 ? "new DefaultEndpointFilterInvocationContext" : "EndpointFilterInvocationContext.Create";
-        var invocationGenericArgs = endpoint.Parameters.Length == 0 ? string.Empty : $"<{endpoint.EmitFilterInvocationContextTypeArgs()}>";
+        var invocationCreator = endpoint.Parameters.Length > 8
+            ? "new DefaultEndpointFilterInvocationContext"
+            : "EndpointFilterInvocationContext.Create";
+        var invocationGenericArgs = endpoint.Parameters.Length is > 0 and < 8
```

Comment:
(nit), it would be nice to have the same condition here instead of flipping them. Maybe even just have a single `if` (but then you couldn't use `var` and that would probably be a major violation in asp.net coding guidelines):

```C#
string invocationCreator;
string invocationGenericArgs;
if (endpoint.Parameters.Length <= 8)
{
    invocationCreator = "EndpointFilterInvocationContext.Create";
    invocationGenericArgs = $"<{endpoint.EmitFilterInvocationContextTypeArgs()}>";
}
else
{
    invocationCreator = "new DefaultEndpointFilterInvocationContext";
    invocationGenericArgs = string.Empty;
}
```

Reply:
Oooh, I see the difference - `> 0`. Maybe just move the `< >` part into `EmitFilterInvocationContextTypeArgs()`, and then that method would skip adding it.

```C#
string invocationCreator;
string invocationGenericArgs;
if (endpoint.Parameters.Length <= 8)
{
    invocationCreator = "EndpointFilterInvocationContext.Create";
    invocationGenericArgs = endpoint.EmitFilterInvocationContextTypeArgs();
}
else
{
    invocationCreator = "new DefaultEndpointFilterInvocationContext";
    invocationGenericArgs = string.Empty;
}

    public static string EmitFilterInvocationContextTypeArgs(this Endpoint endpoint)
    {
        if (endpoint.Parameters.Length == 0)
        {
            return string.Empty;
        }
 
        var sb = new StringBuilder("<");
 
        for (var i = 0; i < endpoint.Parameters.Length; i++)
        {
            sb.Append(endpoint.Parameters[i].Type.ToDisplayString(endpoint.Parameters[i].IsOptional ? NullableFlowState.MaybeNull : NullableFlowState.NotNull, EmitterConstants.DisplayFormat));
 
            if (i < endpoint.Parameters.Length - 1)
            {
                sb.Append(", ");
            }
        }
        
         sb.Append('>');
 
        return sb.ToString();
    }
```